File: julia/tie.jl

function prob(n)
  n = BigFloat(n)
  two = BigFloat(2)
  x = n/2
  a = factorial(n)
  b = factorial(x)
  b *= b
  c = two^n
  return a / b / c
end

for n in [3030, 1000, 500, 200, 100, 50, 10]
  p = float64(round(prob(n)*100, 2))
  println(n, " : ", p, " %")
end

File: ruby/tie.rb
def prob(n)
  x = n/2
  b = (2..x).reduce(:*)
  a = b * ((x+1)..n).reduce(:*)
  b *= b
  c = 2**n
  (a.to_r / b.to_r / c).to_f
end

[3030, 1000, 500, 200, 100, 50, 10].each do |n|
  p = (prob(n)*100).round(2)
  puts "#{n} : #{p} %"
end
